shotover-proxy/tests/redis_int_tests/basic_driver_tests.rs

```diff
@@ -1271,33 +1273,48 @@ pub async fn test_trigger_transform_failure_driver(connection: &mut Connection)
 ///
 /// CAREFUL: This lacks any kind of check that shotover is ready,
 /// so make sure shotover_manager.redis_connection is run on 6379 before calling this.
-pub fn test_trigger_transform_failure_raw() {
+pub async fn test_trigger_transform_failure_raw() {
     // Send invalid redis command
     // To correctly handle this shotover should close the connection
-    let mut connection = std::net::TcpStream::connect("127.0.0.1:6379").unwrap();
-    connection.write_all(b"*1\r\n$4\r\nping\r\n").unwrap();
-    connection
-        .set_read_timeout(Some(Duration::from_secs(10)))
```

Comment:
We've lost the read timeout which is a little concerning.
The test would hang if shotover never replied.
That sounds like quite an easy bug to accidentally introduce to shotover during development and a hung test is a pain to debug.

So probably the easiest fix is to just use https://doc.rust-lang.org/std/net/struct.TcpStream.html#method.connect_timeout to perform a connection timeout on non async TcpStream.
Alternatively we could figure out the alternative for set_read_timeout on `tokio::net::TcpStream`

Reply:
I added the timeout below.

Happy to just switch back to the std TcpStream though.

Reply:
oh duh, you are doing a tokio timeout on the connection.read, that works fine. :+1: